services/statestorage/adapter/memory_persistence.go

```diff
@@ -3,132 +3,109 @@ package adapter
 import (
 	"bytes"
 	"fmt"
+	"github.com/orbs-network/orbs-network-go/services/statestorage/merkle"
 	"github.com/orbs-network/orbs-spec/types/go/primitives"
 	"github.com/orbs-network/orbs-spec/types/go/protocol"
-	"github.com/pkg/errors"
 	"sort"
 	"strings"
+	"sync"
 )
 
-type ContractState map[string]*protocol.StateRecord
-type StateVersion map[primitives.ContractName]ContractState
-
 type InMemoryStatePersistence struct {
-	snapshots map[primitives.BlockHeight]StateVersion
-	roots     map[primitives.BlockHeight]primitives.MerkleSha256
+	mutex       sync.RWMutex
+	bState      ChainState
```

Comment:
why the `b` prefix? please use something more meaningful

Reply:
replaced with prst, as in "persisted"